### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by TakenPilot**
New Feature
- Sites can now set webhooks for 'publish' and 'unpublish' events.

```yaml
host: 'nymag.com'
#...config, etc
notify:
  webhooks:
    published:
      - http://some-url.com/some-path
    unpublished:
      - http://some-url.com/some-path
```

The webhook will be a POST and will set a `X-Event` header with the name of the event, and will include the data from the event.

The data that the events should send has not been made clear yet, so currently is just whatever is available.  Expect that to change later.

### CODE REVIEW

File: test/fixtures/api-accepts.js [hunk 1]
```diff
@@ -416,7 +417,7 @@ function beforeEachTest(options) {
   stubSchema(options.sandbox);
   stubGetTemplate(options.sandbox);
   stubMultiplexRender(options.sandbox);
-  stubLogging(options.sandbox);
+  //stubLogging(options.sandbox);
```
Comment (yoshokatana) [inline test/fixtures/api-accepts.js]: is this supposed to be commented out?

**Comment by yoshokatana** [comment]
:+1:

File: lib/services/notifications.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  if (_.isPlainObject(data)) {
+    headers[contentType] = 'application/json';
+    options.body = JSON.stringify(data);
+  } else if (_.isString(data)) {
+    headers[contentType] = 'text/plain';
+    options.body = data;
+  }
+
+  console.log('callWebhook', event, url, data);
```
Comment (heichwald) [inline lib/services/notifications.js]: curious why we sometimes use console and sometimes log.info

File: lib/services/notifications.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  restler.request(url, options).on('complete', function (result, response) {
+    log.info('called webhook', _.pick(response, ['status', 'statusText']), result);
```
Comment (heichwald) [inline lib/services/notifications.js]: was thinking of maybe using https://github.com/request/request-promise to use promise everywhere ?

File: lib/services/notifications.js (hunk 2 above)
Comment (TakenPilot) [inline lib/services/notifications.js]: Ah, that should be removed.

File: lib/services/notifications.js (hunk 3 above)
Comment (TakenPilot) [inline lib/services/notifications.js]: Restler can be stubbed more easily.

Also, node-fetch is more generic and returns promises already.

File: lib/services/notifications.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ * @param {object} site
+ * @param {string} eventName
+ * @param {*} [data]
+ * @returns {function}
+ */
+function notify(site, eventName, data) {
+  const events = _.get(site, 'notify.webhooks');
+
+  if (events && _.isArray(events[eventName])) {
+    console.log('notify', eventName, events[eventName]);
```
Comment (heichwald) [inline lib/services/notifications.js]: console here too

**Pull request merged**